addons/test_mail/tests/test_mail_composer.py

```diff
@@ -659,6 +659,29 @@ def test_mail_composer_wtpl_complete(self):
         self.assertEqual(set(message.attachment_ids.mapped('res_id')), set(self.test_record.ids))
         self.assertTrue(all(attach not in message.attachment_ids for attach in attachs), 'Should have copied attachments')
 
+    def test_mail_composer_document_based(self):
+        """ Tests a document-based mass mailing with the same address mails
+        This should be allowed and not considered as duplicate in this context
+        """
+        attachment_data = self._generate_attachments_data(2)
+        email_to_1 = 'test.to.1@test.example.com'
+        self.template.write({
+            'auto_delete': False,  # keep sent emails to check content
+            'attachment_ids': [(0, 0, a) for a in attachment_data],
+            'email_to': '%s, %s' % (email_to_1, email_to_1),
```

Comment:
Shouldn't we set the same partner / email on all test_records to clearly mimic the "invoice with same recipient" use case ? Not sure we are not mixing several tests here.